assets/src/amp-story-editor-blocks.js

```diff
@@ -1,28 +1,50 @@
+/**
+ * External dependencies
+ */
+import uuid from 'uuid/v4';
+
 /**
  * WordPress dependencies
  */
 import { addFilter } from '@wordpress/hooks';
 import { compose } from '@wordpress/compose';
 import domReady from '@wordpress/dom-ready';
-import { setDefaultBlockName, getBlockTypes, unregisterBlockType } from '@wordpress/blocks';
-import { select, subscribe } from '@wordpress/data';
+import { getDefaultBlockName, setDefaultBlockName, getBlockTypes, unregisterBlockType } from '@wordpress/blocks';
+import { select, subscribe, dispatch } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { withAttributes, withParentBlock, withBlockName, withHasSelectedInnerBlock, withAmpStorySettings, withAnimationControls } from './components';
 import { ALLOWED_BLOCKS, ALLOWED_CHILD_BLOCKS, BLOCK_TAG_MAPPING } from './constants';
 import { maybeEnqueueFontStyle } from './helpers';
+import { store } from './stores/amp-story';
 
-const { getSelectedBlockClientId, getBlockOrder, getBlocksByClientId } = select( 'core/editor' );
+const { getSelectedBlockClientId, getBlockOrder, getBlocksByClientId, getBlock } = select( 'core/editor' );
+const { getAnimationOrder } = select( 'amp/story' );
+const { addAnimation, removePage } = dispatch( 'amp/story' );
 
-// Ensure that the default block is page when no block is selected.
 domReady( () => {
 	setDefaultBlockName( 'amp/amp-story-page' );
 
 	// Remove all blocks that aren't whitelisted.
 	getBlockTypes().filter( ( { name } ) => ! ALLOWED_BLOCKS.includes( name ) && 'amp/amp-story-page' ).map( ( { name } ) => unregisterBlockType( name ) );
 
+	// Set initial animation order state.
+	getBlocksByClientId( getBlockOrder() )
+		.filter( ( block ) => block.name === 'amp/amp-story-page' )
+		.map( ( page ) => {
```

Comment:
Since the return value is being discarded, why not use a `for...of` loop here?

Reply:
Matter of personal preference I guess :-)

In most cases I prefer using `map` vs. `for` as it's easier to write and grasp, even when not using the return value. Functional programming just works really well in this context IMO.

Looking at this code again, I think this can use `getClientIdsWithDescendants` instead in order to avoid the nested loop. I'll have another look at it.